packages/wrangler/src/pipelines/cli/create.ts

```diff
@@ -29,274 +16,132 @@ export const pipelinesCreateCommand = createCommand({
+				);
+			}
+		} else {
+			// This should never happen due to earlier validation
+			throw new UserError("Either --sql or --sql-file must be provided");
 		}
 
-		if (args.source.length > 0) {
-			const sourceHandlers: Record<string, () => Source> = {
-				http: (): HttpSource => {
-					const http: HttpSource = {
-						type: "http",
-						format: "json",
-						authentication: args.requireHttpAuth,
-					};
-
-					if (args.corsOrigins && args.corsOrigins.length > 0) {
-						http.cors = { origins: args.corsOrigins };
-					}
-
-					return http;
-				},
-				worker: (): BindingSource => ({
-					type: "binding",
-					format: "json",
-				}),
-			};
+		if (!sql) {
+			throw new UserError("SQL query cannot be empty");
+		}
 
-			for (const source of args.source) {
-				const handler = sourceHandlers[source];
-				if (handler) {
-					pipelineConfig.source.push(handler());
+		// Validate SQL before creating pipeline
+		logger.log("🌀 Validating SQL...");
+		try {
+			const validationResult = await validateSql(config, { sql });
+
+			if (
+				validationResult.tables &&
+				Object.keys(validationResult.tables).length > 0
+			) {
+				const tableNames = Object.keys(validationResult.tables);
+				logger.log(
+					`✅ SQL validated successfully. References tables: ${tableNames.join(", ")}`
+				);
+			} else {
+				logger.log("✅ SQL validated successfully.");
+			}
+		} catch (error) {
+			let errorMessage = "Unknown validation error";
+			if (error && typeof error === "object") {
+				const errorObj = error as {
+					notes?: Array<{ text?: string }>;
+					message?: string;
+				};
+				if (
+					errorObj.notes &&
+					Array.isArray(errorObj.notes) &&
+					errorObj.notes[0]?.text
+				) {
+					errorMessage = errorObj.notes[0].text;
+				} else if (error instanceof Error) {
+					errorMessage = error.message;
 				}
 			}
+			throw new UserError(`SQL validation failed: ${errorMessage}`);
 		}
 
-		if (pipelineConfig.source.length === 0) {
-			throw new UserError(
-				"No sources have been enabled. At least one source (HTTP or Worker Binding) should be enabled"
-			);
-		}
-
-		if (args.transformWorker) {
-			pipelineConfig.transforms.push(parseTransform(args.transformWorker));
-		}
+		const pipelineConfig: CreatePipelineRequest = {
+			name: pipelineName,
+			sql,
+		};
 
-		if (args.r2Prefix) {
-			pipelineConfig.destination.path.prefix = args.r2Prefix;
-		}
-		if (args.shardCount) {
-			pipelineConfig.metadata.shards = args.shardCount;
+		logger.log(`🌀 Creating pipeline '${pipelineName}'...`);
+
+		let pipeline;
+		try {
+			pipeline = await createPipeline(config, pipelineConfig);
+		} catch (error) {
+			if (error instanceof APIError && error.code === 10000) {
+				// Show error when no access to v1 Pipelines API
+				throw new UserError(
+					`An authentication error has occurred creating a pipeline with this version of Wrangler. Please try: npm install wrangler@4.36.0`
```

Comment:
This isn't very clear. Could you include the entire command (using `npx`) and explain why the error occurred?